gluecodium/src/main/java/com/here/gluecodium/generator/dart/DartImportResolver.kt

```diff
@@ -116,22 +119,28 @@ internal class DartImportResolver(
 
     private fun resolveExternalImport(
         limeElement: LimeElement,
-        key: String = IMPORT_PATH_NAME
+        key: String,
+        useAlias: Boolean
     ): DartImport? {
         val importPath = (limeElement as? LimeNamedElement)?.external?.dart?.get(key) ?: return null
         val components = importPath.split(':')
+        val alias = if (useAlias) computeAlias(importPath) else null
         return when (components.first()) {
-            "dart" -> DartImport(components.last(), importType = ImportType.SYSTEM)
-            "package" -> DartImport(components.last().removeSuffix(".dart"))
+            "dart" -> DartImport(components.last(), importType = ImportType.SYSTEM, asAlias = alias)
+            "package" -> DartImport(components.last().removeSuffix(".dart"), asAlias = alias)
             else -> DartImport(
                 components.last().removeSuffix(".dart"),
-                importType = ImportType.RELATIVE
+                importType = ImportType.RELATIVE,
+                asAlias = alias
             )
         }
     }
 
     companion object {
         private val collectionSystemImport = DartImport("collection", importType = ImportType.SYSTEM)
         private val collectionPackageImport = DartImport("collection/collection")
+
+        fun computeAlias(importPath: String) =
+            importPath.split(':').last().split('/').last().split('.').first()
```

Comment:
May be add suffix '_' to make it 'internal'?

Reply:
Import aliases are always file-private, regardless of the alias name. Optionally adding a '_' prefix (not suffix) to an alias is just pure decoration. With '_' it looks like a local private variable. Without '_' it looks like a FQN. Syntactically it is neither and the prefix does nothing.

Or did you mean something else?

Reply:
> Import aliases are always file-private, regardless of the alias name

Fair enough, I didn't know